pkg/apis/controlplane/types.go

```diff
@@ -70,6 +70,14 @@ type GroupMember struct {
 	Ports []NamedPort
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// ClusterGroupMembers is a list of GroupMember objects that are currently selected by a ClusterGroup.
+type ClusterGroupMembers struct {
```

Comment:
both ClusterGroupMembers and Group list the same info now. is the plan to use Group for namespaced groups going forward ?

Reply:
I believe I have tried out using Group as the return type for `clustergroupmember` queries, however there was some runtime issues around the lines of `apiserver received an error that is not an metav1.Status` that I cannot get rid of. After quite some experiment I suspect that in the controller api registration, two separate endpoints cannot point to the same struct as storage type (I have not found proof for it in apiserver source code but using a new struct  ClusterGroupMembers solved the problem for me)

Reply:
Sorry, I just noticed that there was also a Group API from the thread, but I didn't find answer from it. Could we reach a consensus about why there are 2 APIs for group member query before merging it?

Reply:
I actually asked you this exact question offline the other day... 
```
wondering if the `clustergroupmembership` (and `groupmembership` API endpoints in the future) is still necessary?
because the same information can essentially be get thru  GET xxx/controlplane/v1beta2/groups/<cgName>  and GET xxx/controlplane/v1beta2/groups/<ns>/<groupName> (where <ns>/<groupName> is the key)  in the future
```
And I thought you wanted to still keep both APIs (in addition to the Group API).
Essentially the Group API exposes the storage for internal Groups, which we want to reuse for ClusterGroups and namespaced Groups. Because of the RBAC concern you mentioned, it makes sense to have two separate APIs for cluster scoped and namespaced Groups. It sounds weird to me though if we want to re-use the `group` endpoint for ClusterGroup membership query and have a new endpoint for namespaced Group queries, since the storage behind `group` is for both. Or maybe we actually don't need `groups` API registered since we have `clustergroupmembership` and `groupmembership` APIs ?

Reply:
i was also thinking about the latter. i.e. no need for `groups` API as it won't be consumed by agents in that fashion and rather utilize the two distinct APIs

Reply:
Removed the 'groups' API endpoint as it's not used

Reply:
still wondering if we need the controlplane.Group resource now that no one is watching it.. but probably defer it

Reply:
We can remove it in a subsequent PR if you think there's no need for `controlplane.Group` and `controlplane.GroupList`

Reply:
@Dyanngg Sorry, I thought we were discussing single API for Group and ClusterGroup or separate APIs for them, that's why I made an example of NetworkPolicyStats.
I agree the consensus you and @abhiraut made here: removing the previous Group API and related structs as they won't be used in that fashion, also agree they can be removed with another PR.